umh-core/pkg/config/manager.go

```diff
@@ -319,39 +348,100 @@ func (m *FileConfigManager) GetConfig(ctx context.Context, tick uint64) (FullCon
 		m.cacheMu.RUnlock()
 		return cfg, nil
 	}
+
+	// File has changed - check if we should start background refresh
+	currentCacheConfig := m.cacheConfig.Clone()
+	hasCache := !m.cacheModTime.IsZero()
 	m.cacheMu.RUnlock()
+
 	// ---------- SLOW PATH (file changed) ----------
+	// Check if a refresh is already in progress using TryLock
+	if m.refreshMu.TryLock() {
+		// Start background refresh only if we have a cached config to return
+		go m.backgroundRefresh(info.ModTime())
+		// Note: mutex will be unlocked in backgroundRefresh
+	}
+
+	// If we have a cached config, return it while background refresh runs
+	if hasCache {
+		return currentCacheConfig, nil
+	}
+
+	// No cached config - fall back to synchronous read (e.g., during tests or init failure)
+	config, rawConfig, err := m.readAndParseConfig(ctx)
+	if err != nil {
+		return FullConfig{}, err
+	}
+
+	// Update cache with the loaded config atomically
+	m.cacheMu.Lock()
+	m.cacheConfig = config
+	m.cacheRawConfig = rawConfig
+	m.cacheModTime = info.ModTime()
+	m.cacheMu.Unlock()
+
+	return config, nil
+}
 
+// backgroundRefresh runs the config refresh logic in a background goroutine
```

Comment:
It is not running in a background go routine. It is instead supposed to be called in a goroutine.